eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -444,7 +422,16 @@ object Router {
 
   case class MultiPartParams(minPartAmount: MilliSatoshi, maxParts: Int)
 
-  case class RouteParams(randomize: Boolean, maxFeeBase: MilliSatoshi, maxFeePct: Double, routeMaxLength: Int, routeMaxCltv: CltvExpiryDelta, ratios: WeightRatios, mpp: MultiPartParams, includeLocalChannelCost: Boolean) {
+  case class RouteParams(randomize: Boolean,
+                         maxFeeBase: MilliSatoshi,
+                         maxFeePct: Double,
+                         routeMaxLength: Int,
+                         routeMaxCltv: CltvExpiryDelta,
+                         ratios: WeightRatios,
+                         mpp: MultiPartParams,
+                         includeLocalChannelCost: Boolean,
+                         experimentName: String,
+                         experimentPercentage: Int) {
```

Comment:
We can't just merge `RouteParams` and `PathFindingConf`, because they do not have the same meaning, as the special handling of `includeLocalChannelCost` shows. `PathFindingConf` is static but `RouteParams` depends on a particular route request.

How about (cc @t-bast interested in your opinion):

```scala
  case class PathFindingConf(randomizeRouteSelection: Boolean,
                             boundaries: SearchBoundaries,
                             weights: WeightRatios,
                             mpp: MultiPartParams)

  // new class that groups some parameters, not sure about Search
  case class SearchBoundaries(maxFeeBase: MilliSatoshi,
                              maxFeePct: Double,
                              maxRouteLength: Int,
                              maxCltv: CltvExpiryDelta)

  case class RouteParams(conf: PathFindingConf,
                         includeLocalChannelCost: Boolean)
```

Reply:
If we go with this, it would make sense to create subsections:

```
    defaults {
      randomize-route-selection = true // when computing a route for a payment we randomize the final selection

      boundaries {
        max-route-length = 6
        max-cltv = 1008
        fee-threshold-sat = 21
        max-fee-pct = 0.03
      }

      ratios {
        base = 0.0
        cltv = 0.05
        channel-age = 0.4
        channel-capacity = 0.55
        hop-cost-base-msat = 500
        hop-cost-millionths = 200
      }

      mpp {
	    min-amount-satoshis = 15000
	    max-parts = 5
      }
    }
```

Reply:
I've changed `PathFindingConf` to be a nested case class as you suggested but I kept `RouteParams` as a separate flat case class because many of its fields are modified in many places and it's not possible to modify nested fields in a nice way.

Reply:
Take a look at `quicklens`:

```scala
import com.softwaremill.quicklens.ModifyPimp
val aliceParams2 = Alice.nodeParams.modify(_.routerConf.requestNodeAnnouncements).setTo(true)
```